torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1496,3 +1533,24 @@ def evaluate_expr(self, expr: "sympy.Expr", hint=None):
                 self.guards.append(
                     ShapeGuard(sympy.Eq(expr, concrete_val), stack))  # type: ignore[arg-type]
         return concrete_val
+
+def _should_allocate(user_marked_dynamic, assume_static_by_default):
+    """
+    Mainly here for readability, repurposes the flag name for the context
+    of shape_env, which cares about allocation.
+    """
+    if user_marked_dynamic:
+        return True
+    # If we got here, the user did *NOT* mark this dim as dynamic,
+    # but BC behavior is to allocate a symbol anyway.
+    return not assume_static_by_default
+
+def _is_dim_dynamic(t, d):
+    return hasattr(t, "_dynamo_dynamic_indices") and d in t._dynamo_dynamic_indices
+
+def _is_int(expr):
+    try:
+        int(expr)
+    except RuntimeError:
+        return False
```

Comment:
This is terrible, can we do this test some other way. If expr is guaranteed to be sympy, you can just look at free symbols

Reply:
Not sure how to do this for when it is a SymInt...

Reply:
Spoke offline ty for the pattern